lib/views/multi-file-patch-view.js

```diff
@@ -97,9 +102,65 @@ export default class MultiFilePatchView extends React.Component {
         this.props.refInitialFocus && this.props.refInitialFocus.setter(element);
       }),
     );
+
+    // Synchronously maintain the editor's scroll position and logical selection across buffer updates.
+    this.suppressChanges = false;
+    let lastScrollTop = null;
+    let lastScrollLeft = null;
+    let lastSelectionIndex = null;
+    if (this.props.onWillUpdatePatch && this.props.onDidUpdatePatch) {
```

Comment:
We could do this more cleanly by specifying default props for these subscription methods. (For example, if this component is rendered with `onWillUpdatePatch` but not `onDidUpdatePatch`, neither will be called, which feels weird.)